saleor/checkout/calculations.py

```diff
@@ -96,18 +109,187 @@ def checkout_line_total(
+        checkout_line.unit_price_net_amount = QP(tax_line_data.unit_net_amount)
+        checkout_line.unit_price_gross_amount = QP(tax_line_data.unit_gross_amount)
+
+        checkout_line.total_price_net_amount = QP(tax_line_data.total_net_amount)
+        checkout_line.total_price_gross_amount = QP(tax_line_data.total_gross_amount)
+
+    CheckoutLine.objects.bulk_update(
+        checkout_lines,
+        [
+            "unit_price_net_amount",
+            "unit_price_gross_amount",
+            "total_price_net_amount",
+            "total_price_gross_amount",
+        ],
+    )
+
+
+def _get_tax_data_from_plugins(
+    checkout: "Checkout",
+    manager: "PluginsManager",
+    checkout_info: "CheckoutInfo",
+    lines: Iterable["CheckoutLineInfo"],
+    address: Optional["Address"],
+    discounts: Optional[Iterable[DiscountInfo]] = None,
+) -> Optional[TaxData]:
+    def get_line_total_price(line_info: "CheckoutLineInfo") -> "TaxedMoney":
+        return manager.calculate_checkout_line_total(
+            checkout_info,
+            lines,
+            line_info,
+            address,
+            discounts or [],
+        )
+
+    def get_line_unit_price(
+        total: "TaxedMoney", line_info: "CheckoutLineInfo"
+    ) -> "TaxedMoney":
+        return manager.calculate_checkout_line_unit_price(
+            total,
+            line_info.line.quantity,
+            checkout_info,
+            lines,
+            line_info,
+            address,
+            discounts or [],
+        )
+
+    try:
+        tax_lines = [
+            TaxLineData(
+                id=0,
+                currency=checkout.currency,
+                total_net_amount=(
+                    total_price := get_line_total_price(line_info)
+                ).net.amount,
+                total_gross_amount=total_price.gross.amount,
+                unit_net_amount=(
+                    unit_price := get_line_unit_price(total_price, line_info)
+                ).net.amount,
+                unit_gross_amount=unit_price.gross.amount,
+            )
+            for line_info in lines
+        ]
+
+        shipping_price = manager.calculate_checkout_shipping(
+            checkout_info, lines, address, discounts or []
+        )
+        subtotal = manager.calculate_checkout_subtotal(
+            checkout_info, lines, address, discounts or []
+        )
+        total = manager.calculate_checkout_total(
+            checkout_info, lines, address, discounts or []
+        )
+
+        return TaxData(
+            currency=checkout.currency,
+            total_net_amount=total.net.amount,
+            total_gross_amount=total.gross.amount,
+            subtotal_net_amount=subtotal.net.amount,
+            subtotal_gross_amount=subtotal.gross.amount,
+            shipping_price_net_amount=shipping_price.net.amount,
+            shipping_price_gross_amount=shipping_price.gross.amount,
+            lines=tax_lines,
+        )
+    except TaxError:
```

Comment:
What will happen here in case of `checkoutComplete` mutation, when tax plugin will raise an TaxError? I believe that this is different behaviour than currently. `fetch_checkout_prices_if_expired` silently won't do anything, which means that it will not calculate any taxes, and checkout will have the old prices, or prices which will not have applied any taxes.